bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryConfig.java

```diff
@@ -65,4 +66,40 @@ public interface BigQueryConfig {
   Optional<Integer> getFlowControlWindowBytes();
 
   Priority getQueryJobPriority();
+
+  default int clientCreationHashCode() {
+    return Objects.hashCode(
+        getAccessTokenProviderFQCN(),
+        getAccessTokenProviderConfig(),
+        getCredentialsKey(),
+        getAccessToken(),
+        getCredentialsFile(),
+        getBigQueryHttpEndpoint(),
+        getFlowControlWindowBytes(),
+        getBigQueryStorageGrpcEndpoint(),
+        getCreateReadSessionTimeoutInSeconds(),
+        getBigQueryProxyConfig(),
+        getParentProjectId(),
```

Comment:
Why do we need the parent project ID ? it is not used by the factory

Reply:
It is used while creating `HeaderProvider` https://github.com/GoogleCloudDataproc/spark-bigquery-connector/blob/9e31d999f8c7e80121b48a32a24d0d28c64189c7/bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/HttpUtil.java#L30
Added just to ensure all the dependencies are equal.